stl/inc/algorithm

```diff
@@ -2214,6 +2217,134 @@ namespace ranges {
     };
 
     inline constexpr _Search_n_fn search_n{_Not_quite_object::_Construct_tag{}};
+
+#if _HAS_CXX23
+    class _Starts_with_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_iterator _It1, sentinel_for<_It1> _Se1, input_iterator _It2, sentinel_for<_It2> _Se2,
+            class _Pr = ranges::equal_to, class _Pj1 = identity, class _Pj2 = identity>
+            requires indirectly_comparable<_It1, _It2, _Pr, _Pj1, _Pj2>
+        _NODISCARD constexpr bool operator()(_It1 _First1, _Se1 _Last1, _It2 _First2, _Se2 _Last2, _Pr _Pred = {},
+            _Pj1 _Proj1 = {}, _Pj2 _Proj2 = {}) const {
+            // clang-format on
+            _Adl_verify_range(_First1, _Last1);
+            _Adl_verify_range(_First2, _Last2);
+
+            auto _UFirst1      = _Get_unwrapped(_STD move(_First1));
+            auto _ULast1       = _Get_unwrapped(_STD move(_Last1));
+            auto _UFirst2      = _Get_unwrapped(_STD move(_First2));
+            const auto _ULast2 = _Get_unwrapped(_STD move(_Last2));
+            if constexpr (sized_sentinel_for<_Se1, _It1> && sized_sentinel_for<_Se2, _It2>) {

```

Comment:
Incidentally that might be one of the use cases for infininte ranges. Iterating over an input stream until it diverges from an expected input and then do some magic. So returning `false` seems definitely the wrong thing in case of two infinite ranges